src/infra/src/table/migration/m20241213_133700_populate_destinations_table.rs

```diff
@@ -0,0 +1,333 @@
+        // records into memory.
+        let mut meta_pages = meta::Entity::find()
+            .filter(meta::Column::Module.eq("destinations"))
+            .order_by_asc(meta::Column::Id)
+            .paginate(&txn, 100);
+
+        while let Some(metas) = meta_pages.fetch_and_next().await? {
+            let new_temp_results: Result<Vec<_>, DbErr> = metas
+                .into_iter()
+                .map(|meta| {
+                    let old_dest: meta_destinations::Destination =
+                        json::from_str(&meta.value).map_err(|e| DbErr::Migration(e.to_string()))?;
+
+                    let new_type = match old_dest.destination_type {
+                        meta_destinations::DestinationType::Http => {
+                            destinations::DestinationType::Http(destinations::Endpoint {
+                                url: old_dest.url,
+                                method: old_dest.method.into(),
+                                skip_tls_verify: old_dest.skip_tls_verify,
+                                headers: old_dest.headers,
+                            })
+                        }
+                        meta_destinations::DestinationType::Email => {
+                            destinations::DestinationType::Email(destinations::Email {
+                                recipients: old_dest.emails,
+                            })
+                        }
+                        meta_destinations::DestinationType::Sns => {
+                            destinations::DestinationType::Sns(destinations::AwsSns {
+                                sns_topic_arn: old_dest.sns_topic_arn.ok_or(DbErr::Migration(
+                                    "SNS destination missing sns_topic_arn".to_string(),
+                                ))?,
+                                aws_region: old_dest.aws_region.ok_or(DbErr::Migration(
+                                    "SNS destination missing aws region info".to_string(),
+                                ))?,
+                            })
+                        }
+                    };
+                    let new_type =
+                        json::to_value(new_type).map_err(|e| DbErr::Migration(e.to_string()))?;
+
+                    let template_id = templates
+                        .get(&format!("{}-{}", meta.key1, old_dest.template))
+                        .cloned();
+                    Ok(destinations::ActiveModel {
+                        id: Set(ider::uuid()),
+                        org: Set(meta.key1),
+                        name: Set(old_dest.name),
+                        module: Set("alert".to_string()), // currently only alerts destinations
+                        template_id: Set(template_id),
+                        pipeline_id: Set(None), // currently only alerts destinations
+                        r#type: Set(new_type),  // this doesn't not have the enum tag!!! fix!
```

Comment:
Just dropping a comment here since you have a `fix` comment on this line

Reply:
ha yes. this was resolved but forgot to remove this comment.